usaspending_api/references/v2/views/cfda.py

```diff
@@ -35,29 +36,37 @@ def get(self, request, cfda=None):
                 "archived": result["archived"],
                 "forecasted": result["forecasted"],
             }
+
         else:
             response = {"results": CFDA_DICTIONARY.values()}
 
         return Response(response)
 
-    def _populate_cfdas_if_needed(self):
-        global CFDA_DICTIONARY
-        if not CFDA_DICTIONARY:
-            response = self._request_from_grants_api()
-
-            #  grants API is brittle in practice, so if we don't get results retry at a polite rate
-            remaining_tries = 30  # 30 attempts two seconds apart gives the max wait time for the API
-            while not response:
-                if remaining_tries == 0:
-                    raise Exception("Failed to get successful response from Grants API!")
-                sleep(2)
-                response = self._request_from_grants_api()
-                remaining_tries = remaining_tries - 1
-
-            CFDA_DICTIONARY = response
-
-    def _request_from_grants_api(self):
-        return post(
-            "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
-            headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
-        ).json()["cfdas"]
+    def get_cfdas(self):
+        cdfas = {}
+        remaining_tries = 10
+
+        while not cdfas and remaining_tries:
+            remaining_tries -= 1
+            response = post(
+                "https://www.grants.gov/grantsws/rest/opportunities/search/cfda/totals",
+                headers={"Authorization": f"APIKEY={settings.GRANTS_API_KEY}"},
+            )
+            if response.status_code != 200:
+                logger.error(f"Status returned by grants API: {response.status_code}")
+                raise NoDataFoundException("Grant data is not currently available.")
+            response = response.json()
+            if response.get("errorMsgs") and response["errorMsgs"] != []:
+                logger.error(f"Error returned by grants API: {response['errorMsgs']}")
+                raise NoDataFoundException("Grant data is not currently available.")
```

Comment:
I'm still not sure it is wise to raise an exception based on `errorMsgs` without even checking if `cfdas` is present and contains acceptable data. Based on the current logic the API response would need to have a HTTP 200 before checking error messages.

Reply:
I would assume that if there is an error from them, any data would be suspect. The API doc (https://www.grants.gov/web/grants/s2s/grantor/restful-apis/opportunity-totals-by-cfda.html) doesn't specify much.

Reply:
> I would assume that if

Yeah, we're on the same page of having uncertainty. I am confused how a HTTP 200 response would contain error messages and not CFDA details. I was thinking if the response is a 200 and provides a list of `cfdas` in the expected schema, then the code could go ahead and provide those results even if `errorMsgs` is populated.

Reply:
So should this return whatever data there is, and pass along the error message to the FE page?

Reply:
I recommend logging the details but not piping the API response directly through our API.

If you prefer to leave the check for `errorMsgs` here and raise an exception that is OK with me. I see your perspective and without understanding the grants.gov api behavior I do not know which assumption is more correct.